mde-services/src/main/java/de/terrestris/mde/mde_backend/service/PublicationService.java

```diff
@@ -315,6 +326,59 @@ public void publishMetadata(String metadataId)
     publishRecords(uuids);
   }
 
+  @PreAuthorize("hasRole('ROLE_MDEADMINISTRATOR') or hasRole('ROLE_MDEEDITOR')")
+  public ResponseEntity<MetadataDeletionResponse> deleteMetadata(
+      String metadataId, Authentication auth) {
+    var metadataCollection =
+        metadataCollectionRepository
+            .findByMetadataId(metadataId)
+            .orElseThrow(() -> new ResponseStatusException(NOT_FOUND));
+    if (metadataCollection.getAssignedUserId() != null) {
+      return new ResponseEntity<>(CONFLICT);
+    }
+    if (!auth.getAuthorities().contains(new SimpleGrantedAuthority("ROLE_MDEADMINISTRATOR"))) {
+      var userId = auth.getName();
+      if (metadataCollection.getOwnerId() == null
+          || metadataCollection.getTeamMemberIds() == null) {
+        return new ResponseEntity<>(CONFLICT);
+      }
+      if (!metadataCollection.getOwnerId().equals(userId)
+          && !metadataCollection.getTeamMemberIds().contains(userId)) {
+        return new ResponseEntity<>(CONFLICT);
+      }
+    }
+
+    var response = new MetadataDeletionResponse();
+    var catalogRecords = new ArrayList<String>();
+
+    List<Service> services = metadataCollection.getIsoMetadata().getServices();
+    if (services != null) {
+      services.forEach(
+          service -> {
+            var fileIdentifier = service.getFileIdentifier();
+            try {
+              removeMetadata(fileIdentifier);
+              catalogRecords.add(fileIdentifier);
+            } catch (Exception e) {
+              log.error(
+                  "Error while removing catalog entry with id {}: \n {}",
+                  fileIdentifier,
+                  e.getMessage());
+              log.trace("Full stack trace: ", e);
+            }
+          });
+    } else {
+      log.warn("No services found for metadata collection with ID {}", metadataId);
+    }
+
+    metadataCollectionRepository.delete(metadataCollection);
+
+    response.setDeletedCatalogRecords(catalogRecords);
+    response.setDeletedMetadataCollection(metadataId);
+
+    return new ResponseEntity<>(response, OK);
```

Comment:
Since this method is available in other service methods, we might want to return a non-response entity value and just throw exceptions where needed.

Reply:
Done